### PR TITLE
`Development`: Rename "Athene" to "Athena"

### PR DESCRIPTION
**Pull request by pal03377**
#### General

- [ ] ~I tested **all** changes and their related features with **all** corresponding user types on a test server.~
- [X] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
None of the checks apply, I just renamed parts of the code
#### Client
None of the checks apply, I just renamed parts of the code

### Motivation and Context

Historically, both names, "Athene" and "Athena" have been used for the semi-automatic (text) assessment system.
We want to be consistent about this. Therefore, we decided to always use the name "Athena" (same as the repository name: https://github.com/ls1intum/Athena) from now on.

As part of migrating the current connection between Artemis and Athena to the new system already found at https://github.com/ls1intum/Athena (#6861), I am also doing a renaming. This is a separate PR so that the actual integration PR will be easier to review.

### Description

I renamed every instance of code with the name "athene" in it to "athena". This includes run configurations. I submitted a PR to https://github.com/ls1intum/artemis-ansible-collection/pull/42 covering this as well.

The only place where the name `athene` is still used is the `de.tum.in.ase.athene` gradle package, which I don't want to rename as it will be removed in a future PR anyways. Therefore, some imports like `import de.tum.in.ase.athene.protobuf.Cluster;` are still using the old name as well.

### Steps for Testing

Unfortunately, you cannot really test the functionality of this PR, as the old Athena system is currently down.
As long as nothing breaks, this is fine, though. In a future PR, the new connection to Athena will be tested anyways and until then, everything remains as-is anyways.

Still, you can
1. Try to find remaining places to rename
2. Deploy this PR to a test server and verify that it generally works
3. Run this PR locally using the `Artemis (Server, Athena)` run configuration to verify that this still generally works as well

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged (only renaming)

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/apollon/ApollonHealthIndicator.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public ApollonHealthIndicator(@Qualifier("shortTimeoutApollonRestTemplate") Rest
     }
 
     /**
-     * Ping Athene at /queueStatus and check if the service is available.
+     * Ping Athena at /queueStatus and check if the service is available.
```
Comment (FelixTJDietrich) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/apollon/ApollonHealthIndicator.java:32]: Should this be Apollon?

**Review by FelixTJDietrich — DISMISSED** [review]
I guess everything related to the `de.tum.in.ase.athene` package can't be renamed right now, you will probably change it during the integration anyways. I did not find any other changes besides the renaming and some import improvements in `AthenaServiceTest.java` (transforming a `.*` import). Also didn't find any other locations to rename, except the Apollon thingy.

Code LGTM, thanks for renaming it for consistency and maintainability!

File: src/main/java/de/tum/in/www1/artemis/service/connectors/apollon/ApollonHealthIndicator.java (hunk 1 above)
Comment (julian-christl) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/apollon/ApollonHealthIndicator.java:32]:
```suggestion
     * Ping Apollon at /queueStatus and check if the service is available.
```

File: src/test/cypress/e2e/exam/ExamAssessment.cy.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ import { Exercise } from '../../support/pageobjects/exam/ExamParticipation';
 let exam: Exam;
 let course: Course;
 
-// This is a workaround for uncaught athene errors. When opening a text submission athene throws an uncaught exception, which fails the test
+// This is a workaround for uncaught athena errors. When opening a text submission athene throws an uncaught exception, which fails the test
```
Comment (julian-christl) [inline src/test/cypress/e2e/exam/ExamAssessment.cy.ts:28]:
```suggestion
// This is a workaround for uncaught athena errors. When opening a text submission athena throws an uncaught exception, which fails the test
```

**Review by julian-christl — CHANGES_REQUESTED** [review]
I only found one more occurrence (apart from the ones mentioned in the PR description)

**Review by FelixTJDietrich — APPROVED** [review]
Code LGTM

**Review by julian-christl — APPROVED** [review]

**Review by MichaelOwenDyer — APPROVED** [review]
LGTM

**Review by maximiliansoelch — APPROVED** [review]

**Comment by krusche** [comment]
Thanks for this contribution. I think this PR is not critical (please correct me if I'm wrong). Due to the necessary config changes, I would postpone merging this PR for another 1-2 weeks as we currently focus on other important bugfixes

**Review by julian-christl — APPROVED** [review]
Reapprove code

**Comment by pal03377** [comment]
This is no longer needed because all changes are included in #6861, which is `maintainer-approved`.

**Pull request closed without merging**